source/common/ur_util.cpp

```diff
@@ -0,0 +1,62 @@
+
+//===--------- ur_adapter_util.cpp - Unified Runtime  ---------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ur_util.hpp"
+#include <cassert>
+
+// Controls tracing UR calls from within the UR itself.
+bool PrintTrace = [] {
+    const char *PiRet = std::getenv("SYCL_PI_TRACE");

```

Comment:
if you change these `std::getenv` calls with `ur_getenv`, you should be able to remove `/D_CRT_SECURE_NO_WARNINGS`.